### PR TITLE
[i18n-ar] Translated file: `docs/source/ar/tasks/summarization.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/tasks/summarization.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### Who can review?
@stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/tasks/summarization.md [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+
+-->
+
+# التلخيص (Summarization)
+
+[[open-in-colab]]
+
+<Youtube id="yHnr5Dk2zCI"/>
+
+يُنشئ التلخيص نسخةً مختصرةً من مستند أو مقال ويلتقط جميع المعلومات المهمة. إلى جانب الترجمة، يُعدّ مثالاً آخر على مهمة يمكن صياغتها كمهمة تسلسل إلى تسلسل. يمكن أن يكون التلخيص:
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:23]:
```suggestion
يقوم التلخيص بإنشاء نسخة مختصرة من مستند أو مقال، حيث يلتقط جميع المعلومات المهمة. بالإضافة إلى الترجمة، يعتبر التلخيص مثالاً آخر على مهمة يمكن صياغتها كتسلسل إلى تسلسل. يمكن أن يكون التلخيص:
```

File: docs/source/ar/tasks/summarization.md [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+
+- استخراجي: استخراج أهم المعلومات من مستند.
+- تجريدي: إنشاء نص جديد يلخص أهم المعلومات.
+
+سيوضح لك هذا الدليل كيفية:
+
+1. ضبط دقيق [T5](https://huggingface.co/google-t5/t5-small) على مجموعة فرعية من مشاريع قوانين ولاية كاليفورنيا من مجموعة بيانات [BillSum](https://huggingface.co/datasets/billsum) للتلخيص التجريدي.
+2. استخدام النموذج المضبوط بدقة للاستدلال.
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:31]:
```suggestion
1. ضبط دقيق [T5](https://huggingface.co/google-t5/t5-small) على مجموعة فرعية من مشاريع قوانين ولاية كاليفورنيا من مجموعة بيانات [BillSum](https://huggingface.co/datasets/billsum) للتلخيص التجريدي.
2. استخدام النموذج المضبوط بدقة للتنبؤ.
```
1.  تدريب نموذج [T5](https://huggingfac

File: docs/source/ar/tasks/summarization.md [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+
+```py
+>>> from huggingface_hub import notebook_login
+
+>>> notebook_login()
+```
+
+## تحميل مجموعة بيانات BillSum
+
+ابدأ بتحميل مجموعة فرعية أصغر من مشاريع قوانين ولاية كاليفورنيا من مجموعة بيانات BillSum من مكتبة 🤗 Datasets:
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:55]:
```suggestion
ابدأ بتحميل جزء صغير من بيانات مشاريع القوانين الخاصة بولاية كاليفورنيا من مجموعة بيانات BillSum في مكتبة 🤗 Datasets:
```

File: docs/source/ar/tasks/summarization.md [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+
+```py
+>>> from datasets import load_dataset
+
+>>> billsum = load_dataset("billsum", split="ca_test")
+```
+
+قسّم مجموعة البيانات إلى مجموعة تدريب ومجموعة اختبار باستخدام طريقة [`~datasets.Dataset.train_test_split`]:
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:63]:
```suggestion
قسّم مجموعة البيانات إلى مجموعتي تدريب واختبار باستخدام الدالة [`~datasets.Dataset.train_test_split`]:
```

File: docs/source/ar/tasks/summarization.md [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+ 'title': 'An act to add Section 10295.35 to the Public Contract Code, relating to public contracts.'}
+```
+
+هناك حقلان سترغب في استخدامهما:
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:78]:
```suggestion
هناك مُدخلان سترغب في استخدامهما:
```

File: docs/source/ar/tasks/summarization.md [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+
+- `text`: نص القانون الذي سيكون مُدخلًا للنموذج.
+- `summary`: نسخة مُختصرة من `text` والتي ستكون هدف النموذج.
+
+## المعالجة المسبقة (Preprocess)
+
+الخطوة التالية هي تحميل مجزء النص T5 لمعالجة `text` و `summary`:
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:85]:
```suggestion
الخطوة التالية هي تحميل مجزء النصوص T5 لمعالجة `text` و `summary`:
```

File: docs/source/ar/tasks/summarization.md [hunk 7]
```diff
@@ -0,0 +1,397 @@
...
+
+```py
+>>> from transformers import AutoTokenizer
+
+>>> checkpoint = "google-t5/t5-small"
+>>> tokenizer = AutoTokenizer.from_pretrained(checkpoint)
+```
+
+دالة المعالجة المسبقة التي تريد إنشاؤها تحتاج إلى:
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:94]:
```suggestion
وظيفة المعالجة المسبقة التي تريد إنشاءها تحتاج إلى:
```

File: docs/source/ar/tasks/summarization.md [hunk 8]
```diff
@@ -0,0 +1,397 @@
...
+
+1. إضافة بادئة للمُدخل بمُطالبة حتى يعرف T5 أن هذه مهمة تلخيص. تتطلب بعض النماذج القادرة على مهام البرمجة اللغوية العصبية المتعددة مطالبات لمهام مُحددة.
+2. استخدام وسيطة الكلمة الرئيسية `text_target` عند ترميز التصنيفات.
+3. اقتطاع التسلسلات بحيث لا يزيد طولها عن الحد الأقصى الذي تم تعيينه بواسطة معامل `max_length`.
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:98]:
```suggestion
1. إضافة بادئة للمُدخل باستخدام توجيه حتى يعرف T5 أن هذه مهمة تلخيص. تتطلب بعض النماذج القادرة على مهام البرمجة اللغوية العصبية المتعددة توجيهات لمهام مُحددة.
2. استخدام مُعامل الكلمة الرئيسية `text_target` عند ترميز التصنيفات.
3. قصّ التسلسلات بحيث لا يزيد طولها عن الحد الأقصى الذي تم تعيينه بواسطة مُعامل `max_length`.
```

File: docs/source/ar/tasks/summarization.md [hunk 9]
```diff
@@ -0,0 +1,397 @@
...
+...     return model_inputs
+```
+
+لتطبيق دالة المعالجة المسبقة على مجموعة البيانات بأكملها، استخدم طريقة [`~datasets.Dataset.map`] الخاصة بـ 🤗 Datasets. يمكنك تسريع دالة `map` عن طريق تعيين `batched=True` لمعالجة عناصر متعددة من مجموعة البيانات في وقت واحد:
+
+```py
+>>> tokenized_billsum = billsum.map(preprocess_function, batched=True)
+```
+
+الآن قم بإنشاء دفعة من الأمثلة باستخدام [`DataCollatorForSeq2Seq`].  الأكثر كفاءة *الحشو الديناميكي* للجمل إلى أطول طول في دفعة أثناء التجميع، بدلاً من حشو مجموعة البيانات بأكملها إلى الحد الأقصى للطول.
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:119]:
```suggestion
الآن قم بإنشاء دفعة من الأمثلة باستخدام [`DataCollatorForSeq2Seq`].  الأكثر كفاءة *الحشو الديناميكي* للجمل إلى أطول طول في دفعة أثناء عملية التجميع، بدلاً من حشو مجموعة البيانات بأكملها إلى الحد الأقصى للطول.
```

File: docs/source/ar/tasks/summarization.md [hunk 10]
```diff
@@ -0,0 +1,397 @@
...
+>>> from transformers import DataCollatorForSeq2Seq
+
+>>> data_collator = DataCollatorForSeq2Seq(tokenizer=tokenizer, model=checkpoint, return_tensors="tf")
+```
+</tf>
+</frameworkcontent>
+
+## التقييم (Evaluate)
+
+يُعد تضمين مقياس أثناء التدريب مفيدًا غالبًا لتقييم أداء نموذجك. يمكنك تحميل طريقة تقييم بسرعة باستخدام مكتبة 🤗 [Evaluate](https://huggingface.co/docs/evaluate/index). لهذه المهمة، قم بتحميل مقياس [ROUGE](https://huggingface.co/spaces/evaluate-metric/rouge) (انظر [الجولة السريعة](https://huggingface.co/docs/evaluate/a_quick_tour) الخاصة بـ 🤗 Evaluate لمعرفة المزيد حول كيفية تحميل وحساب مقياس):
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:142]:
```suggestion
يُعد تضمين مقياس أثناء التدريب مفيدًا غالبًا لتقييم أداء نموذجك. يمكنك تحميل طريقة تقييم بسرعة باستخدام مكتبة 🤗 [Evaluate](https://huggingface.co/docs/evaluate/index). لهذه المهمة، قم بتحميل مقياس [ROUGE](https://huggingface.co/spaces/evaluate-metric/rouge) (راجع [الجولة السريعة](https://huggingface.co/docs/evaluate/a_quick_tour) الخاصة بـ 🤗 Evaluate لمعرفة المزيد حول كيفية تحميل وحساب مقياس):
```

File: docs/source/ar/tasks/summarization.md [hunk 11]
```diff
@@ -0,0 +1,397 @@
...
+>>> from transformers import AutoModelForSeq2SeqLM, Seq2SeqTrainingArguments, Seq2SeqTrainer
+
+>>> model = AutoModelForSeq2SeqLM.from_pretrained(checkpoint)
+```
+
+في هذه المرحلة، لم يتبق سوى ثلاث خطوات:
+
+1. حدد معلمات التدريب الخاصة بك في [`Seq2SeqTrainingArguments`]. المعامل الوحيد المطلوب هو `output_dir` الذي يُحدد مكان حفظ نموذجك. ستدفع هذا النموذج إلى Hub عن طريق تعيين `push_to_hub=True` (تحتاج إلى تسجيل الدخول إلى Hugging Face لتحميل نموذجك). في نهاية كل حقبة، سيقوم [`Trainer`] بتقييم مقياس ROUGE وحفظ نقطة تفتيش التدريب.
+2. مرر وسائط التدريب إلى [`Seq2SeqTrainer`] جنبًا إلى جنب مع النموذج ومجموعة البيانات والمحلل اللغوي وجامع البيانات ودالة `compute_metrics`.
+3. استدعي [`~Trainer.train`] لضبط نموذجك.
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:194]:
```suggestion
1. حدد مُعامِلات التدريب الخاصة بك في [`Seq2SeqTrainingArguments`]. المعامل الوحيد المطلوب هو `output_dir` الذي يُحدد مكان حفظ نموذجك. ستدفع هذا النموذج إلى Hub عن طريق تعيين `push_to_hub=True` (تحتاج إلى تسجيل الدخول إلى Hugging Face لتحميل نموذجك). في نهاية كل حقبة، سيقوم [`Trainer`] بتقييم مقياس ROUGE وحفظ نقطة تفتيش التدريب.
2. مرر مُعامِلات التدريب إلى [`Seq2SeqTrainer`] جنبًا إلى جنب مع النموذج ومجموعة البيانات والمُحلِّل اللغوي وجامع البيانات ودالة `compute_metrics`.
3. استدعِ [`~Trainer.train`] لضبط نموذجك.
```

File: docs/source/ar/tasks/summarization.md [hunk 12]
```diff
@@ -0,0 +1,397 @@
...
+للحصول على مثال أكثر تعمقًا حول كيفية ضبط نموذج للتجميع، ألقِ نظرة على [دفتر ملاحظات PyTorch](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/summarization.ipynb)
+أو [دفتر ملاحظات TensorFlow](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/summarization-tf.ipynb) المقابل.
+
+</Tip>
+
+## الاستدلال (Inference)
+
+رائع، الآن بعد أن قمت بضبط نموذج، يمكنك استخدامه للاستدلال!
+
+ابتكر بعض النص الذي ترغب في تلخيصه. بالنسبة لـ T5، تحتاج إلى إضافة بادئة إلى مُدخلاتك اعتمادًا على المهمة التي تعمل عليها. بالنسبة للتجميع، يجب عليك إضافة بادئة إلى مُدخلاتك كما هو موضح أدناه:
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:326]:
```suggestion
خدد بعض النصوص الذي ترغب في تلخيصها. بالنسبة لـ T5، تحتاج إلى إضافة بادئة إلى مُدخلاتك اعتمادًا على المهمة التي تعمل عليها. بالنسبة التلخيص، يجب عليك إضافة بادئة إلى مُدخلاتك كما هو موضح أدناه:
```

File: docs/source/ar/tasks/summarization.md [hunk 13]
```diff
@@ -0,0 +1,397 @@
...
+
+```py
+>>> text = "summarize: The Inflation Reduction Act lowers prescription drug costs, health care costs, and energy costs. It's the most aggressive action on tackling the climate crisis in American history, which will lift up American workers and create good-paying, union jobs across the country. It'll lower the deficit and ask the ultra-wealthy and corporations to pay their fair share. And no one making under $400,000 per year will pay a penny more in taxes."
+```
+
+أبسط طريقة لتجربة نموذجك المضبوط للاستدلال هي استخدامه في [`pipeline`]. قم بإنشاء مثيل لـ `pipeline` للتجميع باستخدام نموذجك، ومرر نصك إليه:
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:332]:
```suggestion
أبسط طريقة لتجربة نموذجك المضبوط للاستدلال هي استخدامه في [`pipeline`]. استخدم `pipeline` للتلخيص باستخدام نموذجك، ومرر نصك إليه:
```

File: docs/source/ar/tasks/summarization.md [hunk 14]
```diff
@@ -0,0 +1,397 @@
...
+قسم النص وإرجع `input_ids` كتنسورات TensorFlow:
+
+```py
+>>> from transformers import AutoTokenizer
+
+>>> tokenizer = AutoTokenizer.from_pretrained("username/my_awesome_billsum_model")
+>>> inputs = tokenizer(text, return_tensors="tf").input_ids
+```
+
+استخدم طريقة [`~transformers.generation_tf_utils.TFGenerationMixin.generate`] لإنشاء التجميع. لمزيد من التفاصيل حول استراتيجيات توليد النص المختلفة والمعلمات للتحكم في التوليد، راجع واجهة برمجة تطبيقات [توليد النص](../main_classes/text_generation).
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:381]:
```suggestion
استخدم طريقة [`~transformers.generation_tf_utils.TFGenerationMixin.generate`] لإنشاء التلخيص. لمزيد من التفاصيل حول استراتيجيات توليد النص المختلفة والمعلمات للتحكم في التوليد، راجع واجهة برمجة تطبيقات [توليد النص](../main_classes/text_generation).
```

File: docs/source/ar/tasks/summarization.md [hunk 15]
```diff
@@ -0,0 +1,397 @@
...
+قسم النص وإرجع `input_ids` كتنسورات PyTorch:
+
+```py
+>>> from transformers import AutoTokenizer
+
+>>> tokenizer = AutoTokenizer.from_pretrained("username/my_awesome_billsum_model")
+>>> inputs = tokenizer(text, return_tensors="pt").input_ids
+```
+
+استخدم طريقة [`~generation.GenerationMixin.generate`] لإنشاء التجميع. لمزيد من التفاصيل حول استراتيجيات توليد النص المختلفة والمعلمات للتحكم في التوليد، راجع واجهة برمجة تطبيقات [توليد النص](../main_classes/text_generation).
```
Comment (abodacs) [inline docs/source/ar/tasks/summarization.md:355]:
```suggestion
استخدم طريقة [`~generation.GenerationMixin.generate`] لإنشاء التلخيص. لمزيد من التفاصيل حول استراتيجيات توليد النص المختلفة والمعلمات للتحكم في التوليد، راجع واجهة برمجة تطبيقات [توليد النص](../main_classes/text_generation).
```

**Review by abodacs — CHANGES_REQUESTED** [review]
السﻻم عليكم

@AhmedAlmaghz 
تمت المراجعة

جزاك الله خيرا

**Comment by AhmedAlmaghz** [comment]
Thank you @abodacs

**Review by stevhliu — APPROVED** [review]
Thanks for your contribution!
